www/js/diary/infinite_scroll_list.js

```diff
@@ -402,8 +426,12 @@ angular.module('emission.main.diary.infscrolllist',['ui-leaflet',
     }
 
     $scope.populateBasicClasses = function(tripgj) {
-        tripgj.display_start_time = DiaryHelper.getLocalTimeString(tripgj.start_local_dt || tripgj.enter_local_dt);
-        tripgj.display_date = moment((tripgj.start_ts || tripgj.enter_ts) * 1000).format('ddd DD MMM YYYY');
+        if (tripgj.start_local_dt || tripgj.enter_local_dt) {
+          tripgj.display_start_time = DiaryHelper.getLocalTimeString(tripgj.start_local_dt || tripgj.enter_local_dt);
+        } else if (tripgj.end_local_dt || tripgj.exit_local_dt) {
+          tripgj.display_end_time = DiaryHelper.getLocalTimeString(tripgj.end_local_dt || tripgj.exit_local_dt);
+        }
+        tripgj.display_date = moment((tripgj.start_ts || tripgj.enter_ts || tripgj.end_ts || tripgj.exit_ts) * 1000).format('ddd DD MMM YYYY');
```

Comment:
why isn't this just part of the `if (tripgj.end_ts || tripgj.exit_ts)` check on line 435

Reply:
I changed this a bit in https://github.com/e-mission/e-mission-phone/pull/958 as well but it should be working in a more simple way now